api/src/main/java/org/apache/gravitino/rel/Table.java

```diff
@@ -44,6 +44,28 @@
 @Evolving
 public interface Table extends Auditable {
 
+  /**
+   * The reserved property name for the table format. This property indicates the format of the
+   * table, such as "iceberg", "hudi", "lance", etc. Some catalog implementations may use this
+   * property to determine the table format and perform specific operations based on the format.
+   */
+  String PROPERTY_TABLE_FORMAT = "format";
```

Comment:
Shall we also update the reference in `GenericLakehouseTablePropertiesMetadata` ?
https://github.com/apache/gravitino/blob/32d9642e8a656f1a8e7b0f7da71201951a28b865/catalogs/catalog-generic-lakehouse/src/main/java/org/apache/gravitino/catalog/lakehouse/GenericLakehouseTablePropertiesMetadata.java#L50

Reply:
I will do it separately in the follow-up PR.